lib/mail/version_specific/ruby_1_9.rb

```diff
@@ -125,11 +130,7 @@ def Ruby19.b_value_decode(str)
         str = Ruby19.decode_base64(match[2])
         str = charset_encoder.encode(str, charset)
       end
-      decoded = str.encode(Encoding::UTF_8, :undef => :replace, :invalid => :replace, :replace => "")
-      decoded.valid_encoding? ? decoded : decoded.encode(Encoding::UTF_16LE, :invalid => :replace, :replace => "").encode(Encoding::UTF_8)
-    rescue Encoding::UndefinedConversionError, ArgumentError, Encoding::ConverterNotFoundError
-      warn "Encoding conversion failed #{$!}"
-      str.dup.force_encoding(Encoding::UTF_8)
```

Comment:
Note this `rescue` clause also caught exceptions in e.g. `Ruby19.decode_base64` and `charset_encoder.encode` up above.